File: qc-core/src/main/java/com/eager/questioncloud/core/domain/user/repository/UserRepository.java
package com.eager.questioncloud.core.domain.user.repository;

import com.eager.questioncloud.core.domain.user.dto.UserWithCreator;
import com.eager.questioncloud.core.domain.user.model.User;
import com.eager.questioncloud.core.domain.user.vo.AccountType;
import java.util.Optional;

public interface UserRepository {
    User getUserByEmail(String email);

    User getUserByPhone(String phone);

    User getUser(Long uid);

    UserWithCreator getUserWithCreator(Long uid);

    User save(User user);

    Optional<User> getSocialUser(AccountType accountType, String socialUid);

    Boolean checkDuplicatePhone(String phone);

    Boolean checkDuplicateEmail(String email);

    Boolean checkDuplicateSocialUidAndAccountType(String socialUid, AccountType accountType);
}

File: qc-domain/qc-user/qc-user-core/src/main/kotlin/com/eager/questioncloud/user/infrastructure/repository/UserRepository.kt
package com.eager.questioncloud.user.infrastructure.repository

import com.eager.questioncloud.user.domain.User
import com.eager.questioncloud.user.enums.AccountType

interface UserRepository {
    fun getUserByEmail(email: String): User

    fun getUserByPhone(phone: String): User

    fun getUser(uid: Long): User

    fun save(user: User): User

    fun getSocialUser(accountType: AccountType, socialUid: String): User?

    fun checkDuplicatePhone(phone: String): Boolean

    fun checkDuplicateEmail(email: String): Boolean

    fun checkDuplicateSocialUidAndAccountType(socialUid: String, accountType: AccountType): Boolean

    fun findByUidIn(ids: List<Long>): List<User>
}
